internal/integration/unified/entity.go

```diff
@@ -579,135 +603,134 @@ func getKmsCredential(kmsDocument bson.Raw, credentialName string, envVar string
+			if v != nil {
+				provider[e.key] = v
 			}
-
-			awsSecretAccessKey, err := getKmsCredential(aws, "secretAccessKey", "FLE_AWS_SECRET", "")
+		}
+	case "gcp":
+		for _, e := range []struct {
+			key    string
+			envVar string
+		}{
+			{key: "email", envVar: "FLE_GCP_EMAIL"},
+			{key: "privateKey", envVar: "FLE_GCP_PRIVATEKEY"},
+		} {
+			v, err := getKmsCredential(opt, e.key, e.envVar, "")
 			if err != nil {
-				return err
+				return nil, err
 			}
-			if awsSecretAccessKey != "" {
-				kmsProviders["aws"]["secretAccessKey"] = awsSecretAccessKey
+			if v != nil {
+				provider[e.key] = v
 			}
 		}
-
-	}
-
-	if azure, ok := ceo.KmsProviders["azure"]; ok {
-		kmsProviders["azure"] = make(map[string]any)
-
-		azureTenantID, err := getKmsCredential(azure, "tenantId", "FLE_AZURE_TENANTID", "")
+	case "kmip":
+		v, err := getKmsCredential(opt, "endpoint", "", "localhost:5698")
 		if err != nil {
-			return err
+			return nil, err
 		}
-		if azureTenantID != "" {
-			kmsProviders["azure"]["tenantId"] = azureTenantID
+		if v != nil {
+			provider["endpoint"] = v
 		}
-
-		azureClientID, err := getKmsCredential(azure, "clientId", "FLE_AZURE_CLIENTID", "")
+	case "local", "local:name2":
+		v, err := getKmsCredential(opt, "key", "", defaultLocalKeyBase64)
 		if err != nil {
-			return err
+			return nil, err
 		}
-		if azureClientID != "" {
-			kmsProviders["azure"]["clientId"] = azureClientID
-		}
-
-		azureClientSecret, err := getKmsCredential(azure, "clientSecret", "FLE_AZURE_CLIENTSECRET", "")
-		if err != nil {
-			return err
-		}
-		if azureClientSecret != "" {
-			kmsProviders["azure"]["clientSecret"] = azureClientSecret
+		if v != nil {
+			provider["key"] = v
 		}
+	default:
+		return nil, fmt.Errorf("unrecognized KMS provider: %s", key)
 	}
-
-	if gcp, ok := ceo.KmsProviders["gcp"]; ok {
-		kmsProviders["gcp"] = make(map[string]any)
-
-		gcpEmail, err := getKmsCredential(gcp, "email", "FLE_GCP_EMAIL", "")
-		if err != nil {
-			return err
-		}
-		if gcpEmail != "" {
-			kmsProviders["gcp"]["email"] = gcpEmail
-		}
-
-		gcpPrivateKey, err := getKmsCredential(gcp, "privateKey", "FLE_GCP_PRIVATEKEY", "")
-		if err != nil {
-			return err
-		}
-		if gcpPrivateKey != "" {
-			kmsProviders["gcp"]["privateKey"] = gcpPrivateKey
-		}
+	if len(provider) == 0 {
+		return nil, nil
 	}
+	return provider, nil
+}
 
-	if kmip, ok := ceo.KmsProviders["kmip"]; ok {
-		kmsProviders["kmip"] = make(map[string]any)
-
-		kmipEndpoint, err := getKmsCredential(kmip, "endpoint", "", "localhost:5698")
+func (em *EntityMap) addClientEncryptionEntity(entityOptions *entityOptions) error {
+	// Construct KMS providers.
+	kmsProviders := make(map[string]map[string]any)
+	ceo := entityOptions.ClientEncryptionOpts
+	tlsconf := make(map[string]*tls.Config)
+	for key, opt := range ceo.KmsProviders {
+		provider, err := getKmsProvider(key, opt)
 		if err != nil {
 			return err
 		}
-
-		if tlsClientCertificateKeyFile != "" && tlsCAFile != "" {
+		if provider == nil {
```

Comment:
(Optional) For maps, it's conventional to check for `len(provider) == 0` because a `nil` and empty map both report length 0. Consider updating the check here to `if len(provider) == 0` and removing the redundant `if len(provider) == 0` from `getKmsProvider`.